src/src/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -516,12 +517,12 @@ private AuthenticationResult postMessage(String requestMessage, HashMap<String,
 
                 Logger.v(TAG, "Server error message:" + errMessage);
                 if (response.getResponseException() != null) {
-                    throw response.getResponseException();
+                    throw new IOException(response.getResponseException());
```

Comment:
This won't crash, it's not RuntimeException. And the only exception from response is actually IOException (see HttpWebRequest.send). Instead of throwing any exception it's better to be specific